### DISCUSSION TITLE
How does postgres extract auth token from request to identify user so that it can apply row level security?

### DISCUSSION THREAD

**Discussion by roker15** [supabase/supabase#3126] [Questions] [answered]
I see in superbase that it checks row level security against eact request to database. Is this native functionality of postgres or superbase provides it?  To be specific how does postgres identifies which users has made request and then checks RSL against that user? Is this done by database itself or by some separate layer provided by superbase.

**Comment by steve-chavez** [comment, marked as answer]
> To be specific how does postgres identifies which users has made request and then checks RSL against that user?

This is done by decoding the JWT and looking into its role claim to get the corresponding postgres role. All of this is done at the PostgREST authentication layer, see the docs [here](https://postgrest.org/en/v8.0/auth.html#authentication-sequence).
